internal/config/storage.go

```diff
@@ -0,0 +1,43 @@
+package config
+
+import (
+	"github.com/spf13/viper"
+)
+
+type StorageType string
+
+const (
+	DatabaseStorageType = StorageType("database")
+	LocalStorageType    = StorageType("local")
+	GitStorageType      = StorageType("git")
+)
+
+// StorageConfig contains fields which will configure the type of backend in which Flipt will serve
+// flag state.
+type StorageConfig struct {
+	Type  StorageType `json:"type,omitempty" mapstructure:"type"`
+	Local Local       `json:"local,omitempty" mapstructure:"local"`
+	Git   Git         `json:"git,omitempty" mapstructure:"git"`
+}
+
+func (c *StorageConfig) setDefaults(v *viper.Viper) {
+	switch v.GetString("storage.type") {
+	case string(LocalStorageType):
+		v.SetDefault("storage.local.path", ".")
+	case string(GitStorageType):
+		v.SetDefault("storage.git.ref", "main")
+	default:
+		v.SetDefault("storage.type", "database")
+	}
+}
```

Comment:
Could we also add a validation step which ensures that when `storage.type == git` that the `repository` and `ref` is non-empty. As well as path being non-empty for `local` type.

e.g. https://github.com/flipt-io/flipt/blob/main/internal/config/database.go#L72